Balance: Voidwalkers are immune to stun again

jokypoky opened the pull request:
Returns the Voidwalker hardstun immunity. While I would like it to have total stun immunity as it is a being from the void made of glass which is a dielectric, this makes more sense balance wise.

## About The Pull Request

Currently the voidwalker can be HARDSTUNNED, even in space! This PR should return the voidwalker back to having hardstun immunity, while still being slowed by such non lethal devices.

## Why It's Good For The Game

Voidwalkers are beings made of void and glass, with glass being a dielectric, stuns, at least with electrified stun batons, shouldn't too terribly affect then. In relation to the void part, perhaps we can say that it distributes these shocks while immersed in the void. Otherwise I might argue that they should be totally stun immune, but that seems to be a bit much. 


## Changelog



:cl:

balance: Voidwalkers are immune to hardstuns again.

/:cl:

jokypoky committed:
Update voidwalker_mob.dm

Adds Stun immunity for the Voidwalker while in space. While I would like it to have total stun immunity as it is a being from the void made of glass which is a dielectric, this makes more sense balance wise.

RikuTheKiller commented:
I don't see why glass void things should be stunnable to be honest, but you do you. I'd rather they be immune for flavor and balancing reasons.

Voidwalkers have no reason to be caught alive and stunning them is equivalent to killing them. From a flavor standpoint, them being stunned makes no sense.

Voidwalkers should at least be stun resistant even while out of space.

jokypoky commented:
> I don't see why glass void things should be stunnable to be honest, but you do you.
> 
> (I'd rather they be immune for flavor and balancing reasons.)

Agreed, but I thought that if the head fellows thought as much they could edit the PR to do so.

jokypoky commented:
> > Agreed, but I thought that if the head fellows thought as much they could edit the PR to do so.
> 
> Maintainers don't edit PRs. They suggest changes to PRs, and usually only the minimum required for the PR to be approved.
> 
> Minus special exceptions of course.

Thanks for the info!

jokypoky committed:
Update voidwalker_mob.dm

Attempting to fix the silly errors I've encountered.

jokypoky committed:
Update voidwalker_mob.dm

This commit give the Voidwalker FULL stun immunity.

jokypoky committed:
Update voidwalker_mob.dm

Accidentally had an AddElement variable twice

jokypoky committed:
Update voidwalker_mob.dm

TRAIT_NOKNOCKDOWN, not required, as the voidwalker cannot be knockeddown

jokypoky committed:
Update voidwalker_mob.dm

Properly added the trait to the correct "position" in code

jokypoky committed:
Update voidwalker_mob.dm

ADD_TRAIT had too many variables so I had to create another

jokypoky committed:
Balance: Voidwalkers are Immune to stun again

Adds 
status_flags &= ~CANSTUN
to Initialize() and sets 
STAMINA = 0
in damage coefficients.

jokypoky committed:
Merge branch 'master' into patch-1

Time-Green commented:
You didn’t really give a gameplay reason for this, which should generally be nr 1 for balancing. 

Do you know how voidwalker interacts with stuns and stamine damage without this change?

jokypoky committed:
Balance: Voidwalkers are immune to stun again

Adds 
status_flags &= ~CANSTUN 
to Voidwalker to allow for stun immunity.
Also sets 
STAMINA = 0
in damage_coeff()

jokypoky committed:
Merge branch 'master' into patch-1

jokypoky committed:
Merge branch 'master' into patch-1

jokypoky committed:
Merge branch 'master' into patch-1

jokypoky committed:
Merge branch 'master' into patch-1

RikuTheKiller commented:
> I mean anyone who makes a combat oriented change ought to be considering stuns so I am having a hard time swallowing that pill that the author just... didnt.

It's a basicmob thing. It was only recently that basicmobs became able to be stunned at all.

Now stunning makes plenty of sense for a lot of basicmobs, like goats, cows, maybe spiders, rats, carps, bears, whatever. Flesh and blood, decent health pools, etc.

But then you have stuff like voidwalkers, space dragons, blood worms, lavaland mobs, etc. You reach a territory where stuns make zero sense from either a gameplay, balancing or flavor perspective. Sometimes all three.

Inherently, when a basicmob is stunned, it has to be treated similarly to or the same as them being killed. They don't have a chance to be detained, sacrificed or converted. The only reason why you stun a basicmob is to kill it.

And because we didn't even have this capability before a while ago, basicmob code doesn't usually include stun resistances or immunities. Frankly, every antag that is intended to have decent staying power needs some way to deal with stuns. And the main carbon cast do. Basicmobs not so much.

I did take stuns into account for blood worms though. They're fully immune to stuns while outside of a host, but very vulnerable to them while inside of a host. They can also melt restraints off their hosts. That's a product of them being made after basicmobs became stunnable.

jokypoky committed:
Returns Voidwalker Hardstun Immunity

Adds 
status_flags &= ~CANSTUN
while keeping 
STAMINA = 1
in the damage_coeffs

This SHOULD allow for the Voidwalker to be stunnable, without allowing for a hardstun to occur!

jokypoky committed:
Merge branch 'master' into patch-1

ExcessiveUseOfCobblestone committed:
Apply suggestions from code review
